// @radix-ui/themes/dist/cjs/helpers/props/layout.props.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.withLayoutProps = exports.extractLayoutProps = exports.layoutPropDefs = exports.withPaddingProps = exports.extractPaddingProps = exports.paddingPropDefs = void 0;
const breakpoints_1 = require("../breakpoints");
const paddingValues = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
const paddingPropDefs = {
    p: { type: 'enum', values: paddingValues, default: undefined, responsive: true },
    px: { type: 'enum', values: paddingValues, default: undefined, responsive: true },
    py: { type: 'enum', values: paddingValues, default: undefined, responsive: true },
    pt: { type: 'enum', values: paddingValues, default: undefined, responsive: true },
    pr: { type: 'enum', values: paddingValues, default: undefined, responsive: true },
    pb: { type: 'enum', values: paddingValues, default: undefined, responsive: true },
    pl: { type: 'enum', values: paddingValues, default: undefined, responsive: true },
};
exports.paddingPropDefs = paddingPropDefs;
function extractPaddingProps(props) {
    const { p = layoutPropDefs.p.default, px = layoutPropDefs.px.default, py = layoutPropDefs.py.default, pt = layoutPropDefs.pt.default, pr = layoutPropDefs.pr.default, pb = layoutPropDefs.pb.default, pl = layoutPropDefs.pl.default, ...rest } = props;
    return { p, px, py, pt, pr, pb, pl, rest };
}
exports.extractPaddingProps = extractPaddingProps;
function withPaddingProps(props) {
    return [
        (0, breakpoints_1.withBreakpoints)(props.p, 'rt-r-p'),
        (0, breakpoints_1.withBreakpoints)(props.px, 'rt-r-px'),
        (0, breakpoints_1.withBreakpoints)(props.py, 'rt-r-py'),
        (0, breakpoints_1.withBreakpoints)(props.pt, 'rt-r-pt'),
        (0, breakpoints_1.withBreakpoints)(props.pr, 'rt-r-pr'),
        (0, breakpoints_1.withBreakpoints)(props.pb, 'rt-r-pb'),
        (0, breakpoints_1.withBreakpoints)(props.pl, 'rt-r-pl'),
    ]
        .filter(Boolean)
        .join(' ');
}
exports.withPaddingProps = withPaddingProps;
const positionValues = ['static', 'relative', 'absolute', 'fixed', 'sticky'];
const positionEdgeValues = ['auto', '0', '50%', '100%'];
// prettier-ignore
const widthHeightValues = ['auto', 'min-content', 'max-content', '100%', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
const flexShrinkValues = ['0', '1'];
const flexGrowValues = ['0', '1'];
const layoutPropDefs = {
    ...paddingPropDefs,
    position: { type: 'enum', values: positionValues, default: undefined, responsive: true },
    inset: { type: 'enum', values: positionEdgeValues, default: undefined, responsive: true },
    top: { type: 'enum', values: positionEdgeValues, default: undefined, responsive: true },
    right: { type: 'enum', values: positionEdgeValues, default: undefined, responsive: true },
    bottom: { type: 'enum', values: positionEdgeValues, default: undefined, responsive: true },
    left: { type: 'enum', values: positionEdgeValues, default: undefined, responsive: true },
    width: { type: 'enum', values: widthHeightValues, default: undefined, responsive: true },
    height: { type: 'enum', values: widthHeightValues, default: undefined, responsive: true },
    shrink: { type: 'enum', values: flexShrinkValues, default: undefined, responsive: true },
    grow: { type: 'enum', values: flexGrowValues, default: undefined, responsive: true },
};
exports.layoutPropDefs = layoutPropDefs;
function extractLayoutProps(props) {
    const { rest: paddingRest, ...paddingProps } = extractPaddingProps(props);
    const { position = layoutPropDefs.position.default, width = layoutPropDefs.width.default, height = layoutPropDefs.height.default, inset = layoutPropDefs.inset.default, top = layoutPropDefs.top.default, bottom = layoutPropDefs.bottom.default, left = layoutPropDefs.left.default, right = layoutPropDefs.right.default, shrink = layoutPropDefs.shrink.default, grow = layoutPropDefs.grow.default, ...rest } = paddingRest;
    return {
        ...paddingProps,
        position,
        width,
        height,
        inset,
        top,
        bottom,
        left,
        right,
        shrink,
        grow,
        rest,
    };
}
exports.extractLayoutProps = extractLayoutProps;
function withLayoutProps(props) {
    return [
        withPaddingProps(props),
        (0, breakpoints_1.withBreakpoints)(props.position, 'rt-r-position'),
        (0, breakpoints_1.withBreakpoints)(props.shrink, 'rt-r-fs'),
        (0, breakpoints_1.withBreakpoints)(props.grow, 'rt-r-fg'),
        (0, breakpoints_1.withBreakpoints)(props.width, 'rt-r-w'),
        (0, breakpoints_1.withBreakpoints)(props.height, 'rt-r-h'),
        (0, breakpoints_1.withBreakpoints)(props.inset, 'rt-r-inset'),
        (0, breakpoints_1.withBreakpoints)(props.top, 'rt-r-top'),
        (0, breakpoints_1.withBreakpoints)(props.bottom, 'rt-r-bottom'),
        (0, breakpoints_1.withBreakpoints)(props.left, 'rt-r-left'),
        (0, breakpoints_1.withBreakpoints)(props.right, 'rt-r-right'),
    ]
        .filter(Boolean)
        .join(' ');
}
exports.withLayoutProps = withLayoutProps;
