// @radix-ui/themes/dist/cjs/helpers/props/margin.props.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.withMarginProps = exports.extractMarginProps = exports.marginPropDefs = void 0;
const breakpoints_1 = require("../breakpoints");
// prettier-ignore
const marginValues = ['auto', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '-1', '-2', '-3', '-4', '-5', '-6', '-7', '-8', '-9'];
const marginPropDefs = {
    m: { type: 'enum', values: marginValues, default: undefined, responsive: true },
    mx: { type: 'enum', values: marginValues, default: undefined, responsive: true },
    my: { type: 'enum', values: marginValues, default: undefined, responsive: true },
    mt: { type: 'enum', values: marginValues, default: undefined, responsive: true },
    mr: { type: 'enum', values: marginValues, default: undefined, responsive: true },
    mb: { type: 'enum', values: marginValues, default: undefined, responsive: true },
    ml: { type: 'enum', values: marginValues, default: undefined, responsive: true },
};
exports.marginPropDefs = marginPropDefs;
function extractMarginProps(props) {
    const { m = marginPropDefs.m.default, mx = marginPropDefs.mx.default, my = marginPropDefs.my.default, mt = marginPropDefs.mt.default, mr = marginPropDefs.mr.default, mb = marginPropDefs.mb.default, ml = marginPropDefs.ml.default, ...rest } = props;
    return { m, mx, my, mt, mr, mb, ml, rest };
}
exports.extractMarginProps = extractMarginProps;
function withMarginProps(props) {
    return [
        (0, breakpoints_1.withBreakpoints)(props.m, 'rt-r-m'),
        (0, breakpoints_1.withBreakpoints)(props.mx, 'rt-r-mx'),
        (0, breakpoints_1.withBreakpoints)(props.my, 'rt-r-my'),
        (0, breakpoints_1.withBreakpoints)(props.mt, 'rt-r-mt'),
        (0, breakpoints_1.withBreakpoints)(props.mr, 'rt-r-mr'),
        (0, breakpoints_1.withBreakpoints)(props.mb, 'rt-r-mb'),
        (0, breakpoints_1.withBreakpoints)(props.ml, 'rt-r-ml'),
    ]
        .filter(Boolean)
        .join(' ');
}
exports.withMarginProps = withMarginProps;
