// @radix-ui/themes/dist/cjs/helpers/radix-colors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.radixGetMatchingGrayScale = exports.radixGrayScales = exports.radixGrayScalesDesaturated = exports.radixGrayScalePure = exports.radixColorScales = exports.radixColorScalesMetal = exports.radixColorScalesBright = exports.radixColorScalesRegular = void 0;
// prettier-ignore
const radixColorScalesRegular = ['tomato', 'red', 'ruby', 'crimson', 'pink', 'plum', 'purple', 'violet', 'iris', 'indigo', 'blue', 'cyan', 'teal', 'jade', 'green', 'grass', 'brown', 'orange'];
exports.radixColorScalesRegular = radixColorScalesRegular;
const radixColorScalesBright = ['sky', 'mint', 'lime', 'yellow', 'amber'];
exports.radixColorScalesBright = radixColorScalesBright;
const radixColorScalesMetal = ['gold', 'bronze'];
exports.radixColorScalesMetal = radixColorScalesMetal;
// prettier-ignore
const radixColorScales = [...radixColorScalesRegular, ...radixColorScalesBright, ...radixColorScalesMetal];
exports.radixColorScales = radixColorScales;
const radixGrayScalePure = 'gray';
exports.radixGrayScalePure = radixGrayScalePure;
const radixGrayScalesDesaturated = ['mauve', 'slate', 'sage', 'olive', 'sand'];
exports.radixGrayScalesDesaturated = radixGrayScalesDesaturated;
const radixGrayScales = [radixGrayScalePure, ...radixGrayScalesDesaturated];
exports.radixGrayScales = radixGrayScales;
function radixGetMatchingGrayScale(colorScale) {
    switch (colorScale) {
        case 'tomato':
        case 'red':
        case 'ruby':
        case 'crimson':
        case 'pink':
        case 'plum':
        case 'purple':
        case 'violet':
            return 'mauve';
        case 'iris':
        case 'indigo':
        case 'blue':
        case 'sky':
        case 'cyan':
            return 'slate';
        case 'teal':
        case 'jade':
        case 'mint':
        case 'green':
            return 'sage';
        case 'grass':
        case 'lime':
            return 'olive';
        case 'yellow':
        case 'amber':
        case 'orange':
        case 'brown':
        case 'gold':
        case 'bronze':
            return 'sand';
    }
}
exports.radixGetMatchingGrayScale = radixGetMatchingGrayScale;
