// @radix-ui/themes/dist/esm/theme-options.js
import { radixColorScalesRegular, radixColorScalesBright, radixColorScalesMetal, radixColorScales, 
//
radixGrayScalePure, radixGrayScalesDesaturated, radixGrayScales, 
//
radixGetMatchingGrayScale, } from './helpers/radix-colors';
const appearances = ['inherit', 'light', 'dark'];
const accentColors = [...radixColorScales, 'gray'];
const grayColors = [...radixGrayScales, 'auto'];
const panelBackgrounds = ['solid', 'translucent'];
const radii = ['none', 'small', 'medium', 'large', 'full'];
const scalings = ['90%', '95%', '100%', '105%', '110%'];
const themePropDefs = {
    hasBackground: { type: 'boolean', default: true },
    appearance: { type: 'enum', values: appearances, default: 'inherit' },
    accentColor: { type: 'enum', values: accentColors, default: 'indigo' },
    grayColor: { type: 'enum', values: grayColors, default: 'auto' },
    panelBackground: { type: 'enum', values: panelBackgrounds, default: 'translucent' },
    radius: { type: 'enum', values: radii, default: 'medium' },
    scaling: { type: 'enum', values: scalings, default: '100%' },
};
const themeAccentColorsGrouped = [
    { label: 'Regulars', values: [...radixColorScalesRegular] },
    { label: 'Brights', values: [...radixColorScalesBright] },
    { label: 'Metals', values: [...radixColorScalesMetal] },
    { label: 'Gray', values: ['gray'] },
];
const themeAccentColorsOrdered = [
    'gray',
    'gold',
    'bronze',
    'brown',
    'yellow',
    'amber',
    'orange',
    'tomato',
    'red',
    'ruby',
    'crimson',
    'pink',
    'plum',
    'purple',
    'violet',
    'iris',
    'indigo',
    'blue',
    'cyan',
    'teal',
    'jade',
    'green',
    'grass',
    'lime',
    'mint',
    'sky',
];
const themeGrayColorsGrouped = [
    { label: 'Pure', values: [radixGrayScalePure] },
    { label: 'Desaturated', values: ['auto', ...radixGrayScalesDesaturated] },
];
function getMatchingGrayColor(accentColor) {
    if (accentColor === 'gray')
        return 'gray';
    return radixGetMatchingGrayScale(accentColor);
}
export { themePropDefs, getMatchingGrayColor, 
//
themeAccentColorsGrouped, themeAccentColorsOrdered, themeGrayColorsGrouped, };
