// @radix-ui/themes/dist/cjs/components/hover-card.js
"use strict";
'use client';
Object.defineProperty(exports, "__esModule", { value: true });
exports.HoverCardContent = exports.HoverCardTrigger = exports.HoverCardRoot = exports.HoverCard = void 0;
const tslib_1 = require("tslib");
const React = tslib_1.__importStar(require("react"));
const classnames_1 = tslib_1.__importDefault(require("classnames"));
const HoverCardPrimitive = tslib_1.__importStar(require("@radix-ui/react-hover-card"));
const hover_card_props_1 = require("./hover-card.props");
const helpers_1 = require("../helpers");
const theme_1 = require("../theme");
const HoverCardRoot = (props) => (React.createElement(HoverCardPrimitive.Root, { closeDelay: 150, openDelay: 200, ...props }));
exports.HoverCardRoot = HoverCardRoot;
HoverCardRoot.displayName = 'HoverCardRoot';
const HoverCardTrigger = React.forwardRef((props, forwardedRef) => (React.createElement(HoverCardPrimitive.Trigger, { ref: forwardedRef, className: (0, classnames_1.default)('rt-HoverCardTrigger', props.className), ...props, asChild: true })));
exports.HoverCardTrigger = HoverCardTrigger;
HoverCardTrigger.displayName = 'HoverCardTrigger';
const HoverCardContent = React.forwardRef((props, forwardedRef) => {
    const { className, forceMount, container, size = hover_card_props_1.hoverCardContentPropDefs.size.default, ...contentProps } = props;
    return (React.createElement(HoverCardPrimitive.Portal, { container: container, forceMount: forceMount },
        React.createElement(theme_1.Theme, { asChild: true },
            React.createElement(HoverCardPrimitive.Content, { align: "start", sideOffset: 8, collisionPadding: 10, ...contentProps, ref: forwardedRef, className: (0, classnames_1.default)('rt-PopperContent', 'rt-HoverCardContent', className, (0, helpers_1.withBreakpoints)(size, 'rt-r-size')) }))));
});
exports.HoverCardContent = HoverCardContent;
HoverCardContent.displayName = 'HoverCardContent';
const HoverCard = Object.assign({}, {
    Root: HoverCardRoot,
    Trigger: HoverCardTrigger,
    Content: HoverCardContent,
});
exports.HoverCard = HoverCard;
