// @radix-ui/themes/dist/cjs/components/text-field.js
"use strict";
'use client';
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextFieldInput = exports.TextFieldSlot = exports.TextFieldRoot = exports.TextField = void 0;
const tslib_1 = require("tslib");
const React = tslib_1.__importStar(require("react"));
const classnames_1 = tslib_1.__importDefault(require("classnames"));
const primitive_1 = require("@radix-ui/primitive");
const text_field_props_1 = require("./text-field.props");
const helpers_1 = require("../helpers");
const TextFieldContext = React.createContext(undefined);
const TextFieldRoot = React.forwardRef((props, forwardedRef) => {
    const { rest: marginRest, ...marginProps } = (0, helpers_1.extractMarginProps)(props);
    const { children, className, size = text_field_props_1.textFieldPropDefs.size.default, variant = text_field_props_1.textFieldPropDefs.variant.default, color = text_field_props_1.textFieldPropDefs.color.default, radius = text_field_props_1.textFieldPropDefs.radius.default, ...rootProps } = marginRest;
    return (React.createElement("div", { "data-radius": radius, ...rootProps, ref: forwardedRef, className: (0, classnames_1.default)('rt-TextFieldRoot', className, (0, helpers_1.withMarginProps)(marginProps)), onPointerDown: (0, primitive_1.composeEventHandlers)(rootProps.onPointerDown, (event) => {
            const target = event.target;
            if (target.closest('input, button, a'))
                return;
            const input = event.currentTarget.querySelector('.rt-TextFieldInput');
            if (!input)
                return;
            const position = input.compareDocumentPosition(target);
            const targetIsBeforeInput = (position & Node.DOCUMENT_POSITION_PRECEDING) !== 0;
            const cursorPosition = targetIsBeforeInput ? 0 : input.value.length;
            requestAnimationFrame(() => {
                input.setSelectionRange(cursorPosition, cursorPosition);
                input.focus();
            });
        }) },
        React.createElement(TextFieldContext.Provider, { value: { size, variant, color, radius } }, children)));
});
exports.TextFieldRoot = TextFieldRoot;
TextFieldRoot.displayName = 'TextFieldRoot';
const TextFieldSlot = React.forwardRef((props, forwardedRef) => {
    const { rest: paddingRest, ...paddingProps } = (0, helpers_1.extractPaddingProps)(props);
    const { className, color = text_field_props_1.textFieldSlotPropDefs.color.default, gap = text_field_props_1.textFieldSlotPropDefs.gap.default, ...slotProps } = paddingRest;
    const context = React.useContext(TextFieldContext);
    return (React.createElement("div", { "data-accent-color": color, ...slotProps, ref: forwardedRef, className: (0, classnames_1.default)('rt-TextFieldSlot', className, (0, helpers_1.withBreakpoints)(context === null || context === void 0 ? void 0 : context.size, 'rt-r-size'), (0, helpers_1.withBreakpoints)(gap, 'rt-r-gap'), (0, helpers_1.withPaddingProps)(paddingProps)) }));
});
exports.TextFieldSlot = TextFieldSlot;
TextFieldSlot.displayName = 'TextFieldSlot';
const TextFieldInput = React.forwardRef((props, forwardedRef) => {
    var _a, _b, _c, _d;
    const { rest: marginRest, ...marginProps } = (0, helpers_1.extractMarginProps)(props);
    const context = React.useContext(TextFieldContext);
    const hasRoot = context !== undefined;
    const { className, size = (_a = context === null || context === void 0 ? void 0 : context.size) !== null && _a !== void 0 ? _a : text_field_props_1.textFieldPropDefs.size.default, variant = (_b = context === null || context === void 0 ? void 0 : context.variant) !== null && _b !== void 0 ? _b : text_field_props_1.textFieldPropDefs.variant.default, color = (_c = context === null || context === void 0 ? void 0 : context.color) !== null && _c !== void 0 ? _c : text_field_props_1.textFieldPropDefs.color.default, radius = (_d = context === null || context === void 0 ? void 0 : context.radius) !== null && _d !== void 0 ? _d : text_field_props_1.textFieldPropDefs.radius.default, ...inputProps } = marginRest;
    const input = (React.createElement(React.Fragment, null,
        React.createElement("input", { "data-accent-color": color, spellCheck: "false", ...inputProps, ref: forwardedRef, className: (0, classnames_1.default)('rt-TextFieldInput', className, (0, helpers_1.withBreakpoints)(size, 'rt-r-size'), `rt-variant-${variant}`) }),
        React.createElement("div", { "data-accent-color": color, "data-radius": (context === null || context === void 0 ? void 0 : context.radius) ? undefined : radius, className: "rt-TextFieldChrome" })));
    return hasRoot ? (input) : (React.createElement(TextFieldRoot, { ...marginProps, size: size, variant: variant, color: color, radius: radius }, input));
});
exports.TextFieldInput = TextFieldInput;
TextFieldInput.displayName = 'TextFieldInput';
const TextField = Object.assign({}, {
    Root: TextFieldRoot,
    Slot: TextFieldSlot,
    Input: TextFieldInput,
});
exports.TextField = TextField;
