// @radix-ui/themes/dist/esm/components/alert-dialog.js
'use client';
import * as React from 'react';
import classNames from 'classnames';
import * as AlertDialogPrimitive from '@radix-ui/react-alert-dialog';
import { alertDialogContentPropDefs } from './alert-dialog.props';
import { withBreakpoints } from '../helpers';
import { Heading } from './heading';
import { Text } from './text';
import { Theme } from '../theme';
const AlertDialogRoot = (props) => (React.createElement(AlertDialogPrimitive.Root, { ...props }));
AlertDialogRoot.displayName = 'AlertDialogRoot';
const AlertDialogTrigger = React.forwardRef((props, forwardedRef) => React.createElement(AlertDialogPrimitive.Trigger, { ...props, ref: forwardedRef, asChild: true }));
AlertDialogTrigger.displayName = 'AlertDialogTrigger';
const AlertDialogContent = React.forwardRef((props, forwardedRef) => {
    const { className, forceMount, container, size = alertDialogContentPropDefs.size.default, ...contentProps } = props;
    return (React.createElement(AlertDialogPrimitive.Portal, { container: container, forceMount: forceMount },
        React.createElement(Theme, { asChild: true },
            React.createElement(AlertDialogPrimitive.Overlay, { className: "rt-DialogOverlay rt-AlertDialogOverlay" },
                React.createElement(AlertDialogPrimitive.Content, { ...contentProps, ref: forwardedRef, className: classNames('rt-DialogContent', 'rt-AlertDialogContent', className, withBreakpoints(size, 'rt-r-size')) })))));
});
AlertDialogContent.displayName = 'AlertDialogContent';
const AlertDialogTitle = React.forwardRef((props, forwardedRef) => (React.createElement(AlertDialogPrimitive.Title, { asChild: true },
    React.createElement(Heading, { size: "5", mb: "3", trim: "start", ...props, ref: forwardedRef }))));
AlertDialogTitle.displayName = 'AlertDialogTitle';
const AlertDialogDescription = React.forwardRef((props, forwardedRef) => (React.createElement(AlertDialogPrimitive.Description, { asChild: true },
    React.createElement(Text, { as: "p", size: "3", ...props, ref: forwardedRef }))));
AlertDialogDescription.displayName = 'AlertDialogDescription';
const AlertDialogAction = React.forwardRef((props, forwardedRef) => React.createElement(AlertDialogPrimitive.Action, { ...props, ref: forwardedRef, asChild: true }));
AlertDialogAction.displayName = 'AlertDialogAction';
const AlertDialogCancel = React.forwardRef((props, forwardedRef) => React.createElement(AlertDialogPrimitive.Cancel, { ...props, ref: forwardedRef, asChild: true }));
AlertDialogCancel.displayName = 'AlertDialogCancel';
const AlertDialog = Object.assign({}, {
    Root: AlertDialogRoot,
    Trigger: AlertDialogTrigger,
    Content: AlertDialogContent,
    Title: AlertDialogTitle,
    Description: AlertDialogDescription,
    Action: AlertDialogAction,
    Cancel: AlertDialogCancel,
});
export { AlertDialog, AlertDialogRoot, AlertDialogTrigger, AlertDialogContent, AlertDialogTitle, AlertDialogDescription, AlertDialogAction, AlertDialogCancel, };
