// @radix-ui/themes/dist/esm/components/context-menu.js
'use client';
import * as React from 'react';
import classNames from 'classnames';
import * as ContextMenuPrimitive from '@radix-ui/react-context-menu';
import { Slottable } from '@radix-ui/react-slot';
import { ScrollArea } from './scroll-area';
import { contextMenuContentPropDefs, contextMenuItemPropDefs, } from './context-menu.props';
import { withBreakpoints } from '../helpers';
import { Theme, useThemeContext } from '../theme';
import { ThickCheckIcon, ThickChevronRightIcon } from '../icons';
const ContextMenuRoot = (props) => (React.createElement(ContextMenuPrimitive.Root, { ...props }));
ContextMenuRoot.displayName = 'ContextMenuRoot';
const ContextMenuTrigger = React.forwardRef((props, forwardedRef) => React.createElement(ContextMenuPrimitive.Trigger, { ...props, ref: forwardedRef, asChild: true }));
ContextMenuTrigger.displayName = 'ContextMenuTrigger';
const ContextMenuContentContext = React.createContext({});
const ContextMenuContent = React.forwardRef((props, forwardedRef) => {
    const themeContext = useThemeContext();
    const { className, children, size = contextMenuContentPropDefs.size.default, variant = contextMenuContentPropDefs.variant.default, color = contextMenuContentPropDefs.color.default, highContrast = contextMenuContentPropDefs.highContrast.default, container, forceMount, ...contentProps } = props;
    const resolvedColor = color !== null && color !== void 0 ? color : themeContext.accentColor;
    return (React.createElement(ContextMenuPrimitive.Portal, { container: container, forceMount: forceMount },
        React.createElement(Theme, { asChild: true },
            React.createElement(ContextMenuPrimitive.Content, { "data-accent-color": resolvedColor, alignOffset: -Number(size) * 4, collisionPadding: 10, ...contentProps, ref: forwardedRef, className: classNames('rt-PopperContent', 'rt-BaseMenuContent', 'rt-ContextMenuContent', className, withBreakpoints(size, 'rt-r-size'), `rt-variant-${variant}`, { 'rt-high-contrast': highContrast }) },
                React.createElement(ScrollArea, { type: "auto" },
                    React.createElement("div", { className: classNames('rt-BaseMenuViewport', 'rt-ContextMenuViewport') },
                        React.createElement(ContextMenuContentContext.Provider, { value: React.useMemo(() => ({ size, variant, color: resolvedColor, highContrast }), [size, variant, resolvedColor, highContrast]) }, children)))))));
});
ContextMenuContent.displayName = 'ContextMenuContent';
const ContextMenuLabel = React.forwardRef((props, forwardedRef) => (React.createElement(ContextMenuPrimitive.Label, { ...props, ref: forwardedRef, className: classNames('rt-BaseMenuLabel', 'rt-ContextMenuLabel', props.className) })));
ContextMenuLabel.displayName = 'ContextMenuLabel';
const ContextMenuItem = React.forwardRef((props, forwardedRef) => {
    const { className, children, color = contextMenuItemPropDefs.color.default, shortcut, ...itemProps } = props;
    return (React.createElement(ContextMenuPrimitive.Item, { "data-accent-color": color, ...itemProps, ref: forwardedRef, className: classNames('rt-reset', 'rt-BaseMenuItem', 'rt-ContextMenuItem', className) },
        React.createElement(Slottable, null, children),
        shortcut && React.createElement("div", { className: "rt-BaseMenuShortcut rt-ContextMenuShortcut" }, shortcut)));
});
ContextMenuItem.displayName = 'ContextMenuItem';
const ContextMenuGroup = React.forwardRef((props, forwardedRef) => (React.createElement(ContextMenuPrimitive.Group, { ...props, ref: forwardedRef, className: classNames('rt-BaseMenuGroup', 'rt-ContextMenuGroup', props.className) })));
ContextMenuGroup.displayName = 'ContextMenuGroup';
const ContextMenuRadioGroup = React.forwardRef((props, forwardedRef) => (React.createElement(ContextMenuPrimitive.RadioGroup, { ...props, ref: forwardedRef, className: classNames('rt-BaseMenuRadioGroup', 'rt-ContextMenuRadioGroup', props.className) })));
ContextMenuRadioGroup.displayName = 'ContextMenuRadioGroup';
const ContextMenuRadioItem = React.forwardRef((props, forwardedRef) => {
    const { children, className, ...itemProps } = props;
    return (React.createElement(ContextMenuPrimitive.RadioItem, { ...itemProps, ref: forwardedRef, className: classNames('rt-BaseMenuItem', 'rt-BaseMenuRadioItem', 'rt-ContextMenuItem', 'rt-ContextMenuRadioItem', className) },
        React.createElement(Slottable, null, children),
        React.createElement(ContextMenuPrimitive.ItemIndicator, { className: "rt-BaseMenuItemIndicator rt-ContextMenuItemIndicator" },
            React.createElement(ThickCheckIcon, { className: "rt-BaseMenuItemIndicatorIcon rt-ContextMenuItemIndicatorIcon" }))));
});
ContextMenuRadioItem.displayName = 'ContextMenuRadioItem';
const ContextMenuCheckboxItem = React.forwardRef((props, forwardedRef) => {
    const { children, className, shortcut, ...itemProps } = props;
    return (React.createElement(ContextMenuPrimitive.CheckboxItem, { ...itemProps, ref: forwardedRef, className: classNames('rt-BaseMenuItem', 'rt-BaseMenuCheckboxItem', 'rt-ContextMenuItem', 'rt-ContextMenuCheckboxItem', className) },
        React.createElement(Slottable, null, children),
        React.createElement(ContextMenuPrimitive.ItemIndicator, { className: "rt-BaseMenuItemIndicator rt-ContextMenuItemIndicator" },
            React.createElement(ThickCheckIcon, { className: "rt-BaseMenuItemIndicatorIcon rt-ContextMenuItemIndicatorIcon" })),
        shortcut && React.createElement("div", { className: "rt-BaseMenuShortcut rt-ContextMenuShortcut" }, shortcut)));
});
ContextMenuCheckboxItem.displayName = 'ContextMenuCheckboxItem';
const ContextMenuSub = (props) => (React.createElement(ContextMenuPrimitive.Sub, { ...props }));
ContextMenuSub.displayName = 'ContextMenuSub';
const ContextMenuSubTrigger = React.forwardRef((props, forwardedRef) => {
    const { className, children, ...subTriggerProps } = props;
    return (React.createElement(ContextMenuPrimitive.SubTrigger, { ...subTriggerProps, ref: forwardedRef, className: classNames('rt-BaseMenuItem', 'rt-BaseMenuSubTrigger', 'rt-ContextMenuItem', 'rt-ContextMenuSubTrigger', className) },
        React.createElement(Slottable, null, children),
        React.createElement("div", { className: "rt-BaseMenuShortcut rt-ContextMenuShortcut" },
            React.createElement(ThickChevronRightIcon, { className: "rt-BaseMenuSubTriggerIcon rt-ContextMenuSubTriggerIcon" }))));
});
ContextMenuSubTrigger.displayName = 'ContextMenuSubTrigger';
const ContextMenuSubContent = React.forwardRef((props, forwardedRef) => {
    const { className, children, container, forceMount, ...subContentProps } = props;
    const { size, variant, color, highContrast } = React.useContext(ContextMenuContentContext);
    return (React.createElement(ContextMenuPrimitive.Portal, { container: container, forceMount: forceMount },
        React.createElement(Theme, { asChild: true },
            React.createElement(ContextMenuPrimitive.SubContent, { "data-accent-color": color, alignOffset: -Number(size) * 4, 
                // Side offset accounts for the outer solid box-shadow
                sideOffset: 1, collisionPadding: 10, ...subContentProps, ref: forwardedRef, className: classNames('rt-PopperContent', 'rt-BaseMenuContent', 'rt-BaseMenuSubContent', 'rt-ContextMenuContent', 'rt-ContextMenuSubContent', className, withBreakpoints(size, 'rt-r-size'), `rt-variant-${variant}`, { 'rt-high-contrast': highContrast }) },
                React.createElement(ScrollArea, { type: "auto" },
                    React.createElement("div", { className: classNames('rt-BaseMenuViewport', 'rt-ContextMenuViewport') }, children))))));
});
ContextMenuSubContent.displayName = 'ContextMenuSubContent';
const ContextMenuSeparator = React.forwardRef((props, forwardedRef) => (React.createElement(ContextMenuPrimitive.Separator, { ...props, ref: forwardedRef, className: classNames('rt-BaseMenuSeparator', 'rt-ContextMenuSeparator', props.className) })));
ContextMenuSeparator.displayName = 'ContextMenuSeparator';
const ContextMenu = Object.assign({}, {
    Root: ContextMenuRoot,
    Trigger: ContextMenuTrigger,
    Content: ContextMenuContent,
    Label: ContextMenuLabel,
    Item: ContextMenuItem,
    Group: ContextMenuGroup,
    RadioGroup: ContextMenuRadioGroup,
    RadioItem: ContextMenuRadioItem,
    CheckboxItem: ContextMenuCheckboxItem,
    Sub: ContextMenuSub,
    SubTrigger: ContextMenuSubTrigger,
    SubContent: ContextMenuSubContent,
    Separator: ContextMenuSeparator,
});
export { ContextMenu, ContextMenuRoot, ContextMenuTrigger, ContextMenuContent, ContextMenuLabel, ContextMenuItem, ContextMenuGroup, ContextMenuRadioGroup, ContextMenuRadioItem, ContextMenuCheckboxItem, ContextMenuSub, ContextMenuSubTrigger, ContextMenuSubContent, ContextMenuSeparator, };
